### PR TITLE
WinRT activation shim for Microsoft.Management.Deployment

### PR DESCRIPTION
**Pull request by JohnMcPMS**
## Change
Modeled after the similar project for `Microsoft.Management.Configuration`, this new project enables WinRT standard activation via `DllGetActivationFactory` and `IActivationFactory`.  The major change required was to add support for multiple classes.

## Validation
Manually updated the sample caller to use WinRT style activation for the `PackageManager` class and verified that placing the (renamed) .dll next to its .exe resulted in a proper functionality.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4709)

### CODE REVIEW

File: src/Microsoft.Management.Deployment.OutOfProc/PropertySheet.props
Comment (florelis) [inline src/Microsoft.Management.Deployment.OutOfProc/PropertySheet.props:1]: Can probably be deleted

File: src/Microsoft.Management.Deployment.OutOfProc/Microsoft.Management.Deployment.OutOfProc.vcxproj [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="ReleaseStatic|Win32">
+      <Configuration>ReleaseStatic</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="ReleaseStatic|x64">
+      <Configuration>ReleaseStatic</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|ARM64">
```
Comment (florelis) [inline src/Microsoft.Management.Deployment.OutOfProc/Microsoft.Management.Deployment.OutOfProc.vcxproj:40]: We don't have ARM?

File: src/Microsoft.Management.Deployment.OutOfProc/Microsoft.Management.Deployment.OutOfProc.vcxproj [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    </ClCompile>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)'=='Debug'">
+    <ClCompile>
+      <Optimization>Disabled</Optimization>
+      <PreprocessorDefinitions>_DEBUG;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <AdditionalIncludeDirectories Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'">$(ProjectDir)..\AppInstallerSharedLib\Public;$(ProjectDir)..\WinGetServer;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">$(ProjectDir)..\AppInstallerSharedLib\Public;$(ProjectDir)..\WinGetServer;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <TreatWarningAsError Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'">true</TreatWarningAsError>
+      <TreatWarningAsError Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">true</TreatWarningAsError>
```
Comment (florelis) [inline src/Microsoft.Management.Deployment.OutOfProc/Microsoft.Management.Deployment.OutOfProc.vcxproj:199]: Not needed, but I'd love if you could remove all the conditions to keep a single one

**Review by florelis — APPROVED** [review]

File: src/Microsoft.Management.Deployment.OutOfProc/Microsoft.Management.Deployment.OutOfProc.vcxproj (hunk 1 above)
Comment (JohnMcPMS) [reply src/Microsoft.Management.Deployment.OutOfProc/Microsoft.Management.Deployment.OutOfProc.vcxproj:40]: We don't currently ship it in the nuget package this is headed for.

**Pull request merged**